binary_tree_service.py

```diff
@@ -0,0 +1,64 @@
+'''
+Created on 22 May 2017
+
+@author: Eva
+'''
+from binary_tree import BinaryTree
+
+def print_ancestors(tree, key):
+    """Prints the ancestors of a given node.
+    
+    Args:
+        tree: a BinaryTree object that holds all the nodes
+        key: the value of the node
+        
+    Returns:
+        True if the key exists in the binary tree, otherwise false
+    """
+    if tree is None:
+        return False
+    
+    if tree.get_root().get_value() == key:
+        return True
+    
+    if tree.get_root().get_left_node() is not None:
+        left_tree = BinaryTree(tree.get_root().get_left_node())
+        if print_ancestors(left_tree, key):
+            print(tree.get_root().get_value())
+            return True
+            
+    if tree.get_root().get_right_node() is not None:
+        right_tree = BinaryTree(tree.get_root().get_right_node())
+        if print_ancestors(right_tree, key):
+            print(tree.get_root().get_value())
+            return True
+    
+    return False
+
+def find_lowest_common_ancestor(root_node, node1, node2):
+    """Determines the lowest common ancestor of two nodes.
+    
+    Args:
+        root_node: a TreeNode object that is the root of the tree
+        node1: the value of the first node
```

Comment:
Call these `key1` and `key2`, because they are not nodes.
It is often helpful to have similarly named variables (`root_node`, `node1`) by of equivalent types.